dev/ci/README-users.md

```diff
@@ -45,6 +45,25 @@ using [coqbot](https://github.com/coq/bot); meanwhile, a workaround is
 to give merge permissions to someone from the Coq team as to help with
 these kind of merges.
 
+### OCaml and plugin-specific considerations
+
+Developments that link against Coq's OCaml API [most of them are known
+as "plugins"] do have some special requirements:
+
+- Coq's OCaml API is not stable. We hope to improve this in the future
+  but as of today you should expect to have to merge a fair amount of
+  overlays in order to keep your plugin compatible with Coq master.
+
+  In order to alleviate the load, you can delegate the merging of such
+  compatibility pull requests to Coq developers themselves, by
+  granting access to the plugin repository or by using `bots` such as
+  [Bors](https://github.com/apps/bors) that allow for automatic
+  management of Pull Requests.
+
+- Plugins in the CI should compile with the OCaml flags that Coq
+  uses. In particular, warnings that are considered fatal by the Coq
+  developers _must_ be also fatal for plugin CI code.
```

Comment:
I think `should` instead of `must` will be more accurate to what we will actually do.

Reply:
Umm, that's a good question, I am not sure what to think here. Warnings are indeed an issue that causes friction within the whole OCaml community. I am for sure aligned with those on the more strict side, which are a considerable part.

If we view Coq plugins as an extension of Coq then, in the same way we don't allow them in Coq we should not allow them in `master`.

Adding something to the CI is a strong commitment for us, for example I am not very willing to maintain code full of warnings.